src/main/groovy/io/seqera/wave/service/blob/impl/KubeTransferStrategy.groovy

```diff
@@ -61,18 +63,33 @@ class KubeTransferStrategy implements TransferStrategy {
 
     @Override
     BlobCacheInfo transfer(BlobCacheInfo info, List<String> command) {
-        final podName = podName(info)
-        final pod = k8sService.transferContainer(podName, blobConfig.s5Image, command, blobConfig)
-        final terminated = k8sService.waitPod(pod, blobConfig.transferTimeout.toMillis())
-        final stdout = k8sService.logsPod(podName)
-        final result = terminated
-                ? info.completed(terminated.exitCode, stdout)
+        final name = getJobName(info)
+        final job = k8sService.transferJob(name, blobConfig.s5Image, command, blobConfig)
+        final podList = k8sService.waitJob(job, blobConfig.transferTimeout.toMillis())
+        final size = podList?.items?.size() ?: 0
+
+        if( size < 1 )
+            throw new TransferTimeoutException("Transfer job timed out")
+
+        // Find the latest created pod among the pods associated with the job
+        final latestPod = K8sHelper.findLatestPod(podList)
+
+        final pod = k8sService.getPod(latestPod.metadata.name)
+        final exitCode = k8sService.waitPodCompletion(pod, blobConfig.transferTimeout.toMillis())
+        final stdout = k8sService.logsPod(pod)
+
+        final result = exitCode!=null
+                ? info.completed(exitCode, stdout)
                 : info.failed(stdout)
-        cleanupPod(podName, terminated.exitCode)
+
+        // delete job
+        if( cleanup.shouldCleanup(exitCode) && job.metadata?.name ) {
```

Comment:
Is there a specific reason to use use `job.metadata?.name` instead of `name` ?

Reply:
nothing specific, just to make sure its the correct name

Reply:
ok